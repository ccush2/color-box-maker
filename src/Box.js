import React from "react";
import "./index.css";

function Box({ id, width, height, backgroundColor, removeBox }) {
  const handleRemove = () => removeBox(id);

  const style = {
    width: `${width}px`,
    height: `${height}px`,
    backgroundColor,
  };

  return (
    <div className="box" style={style}>
      <button onClick={handleRemove} className="box-remove-btn">
        X
      </button>
    </div>
  );
}

export default Box;
