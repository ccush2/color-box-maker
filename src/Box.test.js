import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Box from "./Box";

// Smoke Test
it("renders without crashing", () => {
  render(<Box width="100" height="100" backgroundColor="blue" />);
});

// Snapshot Test
it("matches snapshot", () => {
  const { asFragment } = render(
    <Box width="100" height="100" backgroundColor="blue" />
  );
  expect(asFragment()).toMatchSnapshot();
});

// Box Removal Test
it('uses the removeBox function when the "X" button is clicked', () => {
  const removeBoxMock = jest.fn();
  const { getByText } = render(
    <Box
      width="100"
      height="100"
      backgroundColor="blue"
      removeBox={removeBoxMock}
    />
  );
  const removeButton = getByText("X");
  fireEvent.click(removeButton);
  expect(removeBoxMock).toHaveBeenCalled();
});
