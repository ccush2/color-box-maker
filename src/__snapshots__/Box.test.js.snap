// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="box"
    style="width: 100px; height: 100px; background-color: blue;"
  >
    <button
      class="box-remove-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
